crates/bevy_pbr/src/render/depth_prepass.rs

```diff
@@ -0,0 +1,786 @@
+        self.entity
+    }
+}
+
+pub struct AlphaMaskDepth3d {
+    pub distance: f32,
+    pub pipeline: CachedRenderPipelineId,
+    pub entity: Entity,
+    pub draw_function: DrawFunctionId,
+}
+
+impl PhaseItem for AlphaMaskDepth3d {
+    type SortKey = FloatOrd;
+
+    #[inline]
+    fn sort_key(&self) -> Self::SortKey {
+        FloatOrd(self.distance)
+    }
+
+    #[inline]
+    fn draw_function(&self) -> DrawFunctionId {
+        self.draw_function
+    }
+}
+
+impl EntityPhaseItem for AlphaMaskDepth3d {
+    #[inline]
+    fn entity(&self) -> Entity {
+        self.entity
+    }
+}
+
+#[derive(Debug, Hash, PartialEq, Eq, Clone, SystemLabel)]
+pub enum Systems {
+    ExtractDepthPhases,
+    QueueDepthPrepassMeshes,
+}
+
+pub struct DepthPrepassPlugin;
+impl Plugin for DepthPrepassPlugin {
+    fn build(&self, app: &mut bevy_app::App) {
+        let render_app = app.sub_app(RenderApp);
+        render_app
+            .add_system_to_stage(
+                RenderStage::Extract,
+                extract_depth_phases.label(Systems::ExtractDepthPhases),
+            )
+            .add_system_to_stage(
+                RenderStage::Queue,
+                queue_depth_prepass_meshes.label(Systems::QueueDepthPrepassMeshes),
+            )
+            .add_system_to_stage(RenderStage::PhaseSort, sort_phase_system::<OpaqueDepth3d>)
+            .add_system_to_stage(
+                RenderStage::PhaseSort,
+                sort_phase_system::<AlphaMaskDepth3d>,
+            )
+            .init_resource::<DepthPrepassMaterialBindGroups>()
+            .init_resource::<DepthPrepassPipeline>()
+            .init_resource::<SpecializedRenderPipelines<DepthPrepassPipeline>>()
+            .init_resource::<DrawFunctions<OpaqueDepth3d>>()
+            .init_resource::<DrawFunctions<AlphaMaskDepth3d>>();
+
+        let depth_prepass_node = DepthPrepassNode::new(&mut render_app.world);
+        render_app.add_render_command::<OpaqueDepth3d, DrawDepth>();
+        render_app.add_render_command::<AlphaMaskDepth3d, DrawDepth>();
+        let render_world = render_app.world.cell();
+        let mut graph = render_world.get_resource_mut::<RenderGraph>().unwrap();
+        let draw_3d_graph = graph
+            .get_sub_graph_mut(bevy_core_pipeline::core_3d::graph::NAME)
+            .unwrap();
+
+        draw_3d_graph.add_node(draw_3d_graph::node::DEPTH_PREPASS, depth_prepass_node);
+        draw_3d_graph
+            .add_node_edge(
+                draw_3d_graph::node::DEPTH_PREPASS,
+                bevy_core_pipeline::core_3d::graph::node::MAIN_PASS,
+            )
+            .unwrap();
+        draw_3d_graph
+            .add_slot_edge(
+                draw_3d_graph.input_node().unwrap().id,
+                bevy_core_pipeline::core_3d::graph::input::VIEW_ENTITY,
+                draw_3d_graph::node::DEPTH_PREPASS,
+                DepthPrepassNode::IN_VIEW,
+            )
+            .unwrap();
+    }
+}
+
+
+// TODO: ActiveCameras
```

Comment:
ActivesCameras are no longer a thing and the user is expected to track that themselves. I've subbed in "all 3d cameras" in the meantime.